File: THCS/2022_2023/CanTho_2223/Bai2/TONGCHAN.cpp
#include <iostream>
#include <fstream>

using namespace std;

int main() {
    // Khai báo biến
    int n;
    long long ai;
    int even_count = 0, odd_count = 0;

    // Đọc dữ liệu từ file TONGCHAN.INP
    ifstream fin("TONGCHAN.INP");
    fin >> n;

    // Đếm số lượng phần tử chẵn và lẻ
    for (int i = 0; i < n; i++) {
        fin >> ai;
        if (ai % 2 == 0) {
            even_count++;
        } else {
            odd_count++;
        }
    }
    fin.close();

    // Kết quả là số lượng phần tử nhỏ hơn giữa chẵn và lẻ
    int result = min(even_count, odd_count);

    // Ghi kết quả ra file TONGCHAN.OUT
    ofstream fout("TONGCHAN.OUT");
    fout << result;
    fout.close();

    return 0;
}

File: THCS/2022_2023/CanTho_2223/Bai2/TONGCHAN.py
# Đọc dữ liệu từ file TONGCHAN.INP
with open('TONGCHAN.INP', 'r') as file:
    # Đọc số lượng phần tử n
    n = int(file.readline().strip())
    
    # Đọc các giá trị của dãy số
    arr = list(map(int, file.readline().split()))

# Đếm số lượng phần tử chẵn và lẻ
even_count = sum(1 for x in arr if x % 2 == 0)
odd_count = n - even_count  # Số lẻ = tổng số phần tử - số chẵn

# Kết quả là số phần tử nhỏ hơn giữa chẵn và lẻ
result = min(even_count, odd_count)

# Ghi kết quả ra file TONGCHAN.OUT
with open('TONGCHAN.OUT', 'w') as file:
    file.write(str(result))
